### PR TITLE
Search new repos bugfix

### PR DESCRIPTION
**Pull request by johnsoncv**
**Bug:**
When a new repository is analyzed there were no results. Attached are two different behaviours, one which is populated (expected), the other is not.
image
image

**Causes for the bug:**
- When a local development environment is first created, it does not have a critical folder for the NLTK library, nltk_data. The fix for this to run the following command to download some library components and create the folder:
`>>> nltk.download()`

- GitHub has changed their url's for comparing branches from 'https://github.com/%s/compare' to https://github.com/%s/compare/master...%s:master. 
compare_changes_crawler has been changed to take those changes into consideration.
- The current method only compares master branches. Further investigation required to compare other branches.

**Important Notes:**
- Running nltk.download() opens a download file. Make sure the download is successful before moving on.
- If you're running the search functionality locally without any local fork information, make sure to change USE_LOCAL_FORK_INFO to False in config.py under DevelopmentConfig.

### CODE REVIEW

File: app/analyse/analyser.py [hunk 1]
```diff
@@ -97,7 +126,8 @@ def start_analyse(repo, access_token):
             },
         )
         repo_forks_list = res.json()
-        localfile_tool.write_to_file(forks_list_path, repo_forks_list)
+        active_forks = get_active_forks(repo,access_token )
```
Comment (LaithHanania) [inline app/analyse/analyser.py]: active_forks should also be passed in project_updater.start_update()

File: app/analyse/analyser.py [hunk 2]
```diff
@@ -48,6 +48,35 @@ def send_mail_for_repo_finish(project_name):
                     username=user.username,
                 )
 
+def get_active_forks(repo, access_token):
+    active_forks = []
+    result_length = 100
+    page = 1
+
+    while result_length == 100:
```
Comment (LaithHanania) [inline app/analyse/analyser.py]: might be good to add a limit to how many pages / active forks to search through so it doesn't take too long for large forks, like add page < 15 in the while condition or len(active_forks) <50

**Review by LaithHanania — CHANGES_REQUESTED** [review]

File: app/analyse/analyser.py (hunk 1 above)
Comment (johnsoncv) [reply app/analyse/analyser.py]: Changed project_updater.start_update() call to use active_forks

File: app/analyse/analyser.py (hunk 2 above)
Comment (johnsoncv) [reply app/analyse/analyser.py]: added a page < 20 limit

**Review by johnperry99 — APPROVED** [review]

**Review by LaithHanania — APPROVED** [review]

**Pull request merged**